pkg/events/queue/queue.go

```diff
@@ -0,0 +1,14 @@
+package queue
+
+import (
+	"github.com/aquasecurity/tracee/pkg/external"
+)
+
+type CacheConfig EventQueue
+
+type EventQueue interface {
+	String() string
+	Setup() error
+	Enqueue(*external.Event)
+	Dequeue(bool) *external.Event // bool is for blocking (true) or non-blocking (false)
```

Comment:
This requires other Queue interfaces to implement Dequeue with a bool which might be unnecessary. I think this should move to be an implementation detail of `EventQueueMem`.